### ISSUE TITLE
Optee cannot store data when both enable CFG_RPMB_FS  AND CFG_REE_FS

### ISSUE THREAD

**Issue by wzhsai** [OP-TEE/optee_os#2759]
I am new to OPTEE. I am running OPTEE on Android Phone.  And enable CFG_RPMB_FS and CFG_REE_FS
In my phone, i have enabled keymaster and gatekeeper.  I have met a problem i cann't store screen lock data to "/data/vendor/tee". After rebooting the device, I cannot unlock the device.
After analysing the problem, I found that the optee cannot find the lock data in "/data/vendor/tee", after rebooting. But if i dont reboot device, optee could find the lock data.

My analysing is bellow:
1, My Android Phone uses FDE. And when the device booting, the keymaster starts early, at this time ,the data partition has not been mounted.
2. Optee will not find the lock data, and it will count the hash data, and store the hash data to RPMB(CFG_RPMB_FS enabled). 
3. After the data partition mounted, the optee using the wrong hash data, so could not find the lock data.

How to solve this problem? Maybe i should use FBE on Android devices?

Thanks

**Comment by jforissier** [comment]
Hi @wzhsai,

You'll need to tell us more. I have tested this configuration on HiKey960 and it worked fine.
Oh, maybe you forgot to set `CFG_RPMB_WRITE_KEY=y`?

**Comment by wzhsai** [comment]
> Hi @wzhsai,
> 
> You'll need to tell us more. I have tested this configuration on HiKey960 and it worked fine.
> Oh, maybe you forgot to set `CFG_RPMB_WRITE_KEY=y`?

Hi @jforissier 

I have add more information. I have set the right RPMB key.  The RPMB partition is working good. There is only the problem i mentioned.
I have also tested that storing the hash data to "REE_FS" not RPMB_FS . And it is ok, after rebooting the phone. My method is that only coment out the MACRO CFG_RPMB_FS in "optee_os/core/tee/tee_ree_fs.c ". How to analyse the Problem? 
Thanks
